===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(proxy_list_slots:slots)) at 53 (line 282, column 4) }
===== Actual patch in diff format =====
@@ -279,7 +279,11 @@ proxy_list_slots (Proxy *py, Mapping *mappings, unsigned int n_mappings)
 
 		if (count > 0) {
 			py->mappings = realloc (py->mappings, sizeof (Mapping) * (py->n_mappings + count));
-			return_val_if_fail (py->mappings != NULL, CKR_HOST_MEMORY);
+			if (py->mappings != NULL) { } else {
+				p11_debug_precond ("p11-kit: '%s' not true at %s\n", "py->mappings != NULL", __func__);
+				free(slots);
+				return CKR_HOST_MEMORY;
+			}
 
 			/* And now add a mapping for each of those slots */
 			for (i = 0; i < count; ++i) {
